cmake/P4ESTConfig.cmake.in

```diff
@@ -0,0 +1,3 @@
+@PACKAGE_INIT@
+
+check_required_components(${PROJECT_NAME})
```

Comment:
@scivision
I think it should be `check_required_components(@PROJECT_NAME@)` instead, so that it is expanded to `check_required_components(P4EST)` otherwise it stays unchanged `check_required_components(${PROJECT_NAME})` and PROJECT_NAME will the value the surrounding scope will provide, which may be something else.

Reply:
also, just to be sure:

- as you know `find_package` can be used in 2 modes (config or module); you provided `FindP4EST.cmake` for the module mode; in this module the logic to find the component (P6EST and P8EST) is there
- but if the client code, issue `find_package(P4EST CONFIG)` the config mode is activated, then P4ESTConfig.cmake is executed, so you need to provide here the same logic to detect components, if not P4EST_${component}_FOUND will never be defined.

I suggest the following modification:
```cmake
@PACKAGE_INIT@

# check for component availability (by import targets and check targets)
include(${CMAKE_CURRENT_LIST_DIR}/@PROJECT_NAME@Targets.cmake)

if(TARGET P4EST::P8EST)
  set(P4EST_P8EST_FOUND true)
endif()

if(TARGET P4EST::P6EST)
  set(P4EST_P6EST_FOUND true)
endif()

check_required_components(@PROJECT_NAME@)
```
Let me know if I'm wrong here.

Reply:
yes I noticed that on several packages that this worked `check_required_components(${PROJECT_NAME})` but not all. The correct syntax is with the double `@` as you noted.

Reply:
yes I corrected the component handling on the last push, thanks